src/Fhir.Anonymizer.Core/Extensions/FhirPathSymbolExtensions.cs

```diff
@@ -7,10 +7,21 @@ namespace Fhir.Anonymizer.Core.Extensions
 {
     public static class FhirPathSymbolExtensions
     {
+        private static object _lock = new object();
         public static SymbolTable AddExtensionSymbols(this SymbolTable t)
         {
-            t.Add("nodesByType", (IEnumerable<ITypedElement> f, string typeName) => NodesByType(f, typeName), doNullProp: true);
-            t.Add("nodesByName", (IEnumerable<ITypedElement> f, string name) => NodesByName(f, name), doNullProp: true);
+            lock (_lock)
+            {
+                if (t.Filter("nodesByType", 2).Count() == 0)
```

Comment:
why here is 2?

Reply:
2 is a required parameter for argumenet count of customized function

Reply:
Maybe we should add some comments

Reply:
Sure, thanks!